integrations/spark/spark-3.1/openhouse-spark-itest/src/test/java/com/linkedin/openhouse/spark/catalogtest/CatalogOperationTest.java

```diff
@@ -68,6 +70,106 @@ public void testCatalogWriteAPI() throws Exception {
     }
   }
 
+  @Test
+  public void testCreateReplicaUnPartitionedTable() throws Exception {
+    try (SparkSession spark = getSparkSession()) {
+      Catalog icebergCatalog = getOpenHouseCatalog(spark);
+      Schema schema =
+          new Schema(
+              Types.NestedField.required(
+                  1,
+                  "a",
+                  Types.StructType.of(Types.NestedField.required(2, "b", Types.StringType.get()))),
+              Types.NestedField.required(3, "c", Types.StringType.get()));
+
+      // Field ids reassigned (Status quo)
+      TableIdentifier tableIdentifier = TableIdentifier.of("replication_test", "t1");
+      Map<String, String> props = new HashMap<>();
+      Table table = icebergCatalog.createTable(tableIdentifier, schema, null, props);
+      Schema schemaAfterCreation = table.schema();
+      Assertions.assertFalse(schemaAfterCreation.sameSchema(schema));
+      Assertions.assertEquals(1, schemaAfterCreation.findField("a").fieldId());
+      Assertions.assertEquals(3, schemaAfterCreation.findField("a.b").fieldId());
+      Assertions.assertEquals(2, schemaAfterCreation.findField("c").fieldId());
```

Comment:
shall we just make it simpler here, by asserting it as false since we expect `a.b` to be 2 and `c` to be 3. 
and the line 101 to 115 can be simplify omitted. basically we will create a case where it failed before the changes but succeeded after the change.